prov/rxm/src/rxm_conn.c

```diff
@@ -1244,11 +1245,29 @@ rxm_conn_handle_reject(struct rxm_ep *rxm_ep, struct rxm_msg_eq_entry *entry)
 	return 0;
 }
 
+static int
+rxm_conn_handle_reset(struct rxm_ep *rxm_ep, struct rxm_msg_eq_entry *entry)
+{
+	struct rxm_cmap_handle *handle = entry->err_entry.fid->context;
+
+	if (handle->state == RXM_CMAP_CONNREQ_SENT) {
+		rxm_conn_close(handle);
+		RXM_CM_UPDATE_STATE(handle, RXM_CMAP_IDLE);
+	} else {
+		FI_WARN(&rxm_prov, FI_LOG_EP_CTRL, "Invalid cmap state: "
+				"%s when receiving connection reset\n",
+				rxm_cm_state_str[handle->state]);
+	}
+	return 0;
+}
+
 static int
 rxm_conn_handle_event(struct rxm_ep *rxm_ep, struct rxm_msg_eq_entry *entry)
 {
 	if (entry->rd == -FI_ECONNREFUSED)
 		return rxm_conn_handle_reject(rxm_ep, entry);
+	if (entry->rd == -FI_ECONNRESET)
+		return rxm_conn_handle_reset(rxm_ep, entry);
```

Comment:
Both of these checks look suspicious.  We should really be checking the entry->event state prior to decoding the error code.

ECONNRESET is a tcp socket specific error code.  It may be best if the tcp provider handled this and returned the correct event based on the connection state (as viewed by rxm).

Reply:
some providers use econnrefused without setting event. I add the same for connection reset due to missing appropriate event, probably we need to add it.